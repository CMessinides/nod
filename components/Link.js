import React from "react";
import PropTypes from "prop-types";
import NextLink from "next/link";
import { createClientRoute, createServerRoute } from "../routes";

function Link({ page, pattern, params = {}, children, ...props } = {}) {
  return (
    <NextLink
      href={createClientRoute(page, params)}
      as={createServerRoute(pattern || page, params)}
      {...props}
    >
      <a>{children}</a>
    </NextLink>
  );
}

Link.propTypes = {
  page: PropTypes.string.isRequired,
  pattern: PropTypes.string,
  params: PropTypes.object,
  children: PropTypes.node.isRequired
};

export default Link;
