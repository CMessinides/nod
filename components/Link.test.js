import React from "react";
import Link from "./Link";
import { render, cleanup } from "@testing-library/react";

afterEach(cleanup);

it("should render", () => {
  const { container } = render(<Link page="/books">All books</Link>);
  expect(container.firstChild).toMatchSnapshot();
});

it("should format a dynamic link with parameters", () => {
  const { container } = render(
    <Link page="/book" pattern="/books/:id" params={{ id: 12 }}>
      Book 12
    </Link>
  );
  expect(container.firstChild).toMatchSnapshot();
});
