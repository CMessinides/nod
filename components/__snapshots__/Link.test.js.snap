// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should format a dynamic link with parameters 1`] = `
<a
  href="/books/12"
>
  Book 12
</a>
`;

exports[`should render 1`] = `
<a
  href="/books"
>
  All books
</a>
`;
